sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3488,18 +3511,28 @@ object functions {
    * Applies a binary operator to an initial state and all elements in the array,
    * and reduces this to a single state. The final state is converted into the final result
    * by applying a finish function.
+   * {{{
+   *   df.select(aggregate(col("i"), lit(0), (acc, x) => acc + x, _ * 10))
+   * }}}
+   *
+   * @param expr the input array column
+   * @param initialValue the initial value
+   * @param merge (combined_value, input_value) => combined_value, the merge function to merge
+   *              an input value to the combined_value
+   * @param finish col => final_col, the lambda function to convert the single state into
```

Comment:
nit: `combined_value => final_value, the lambda function to convert the combined value of all inputs to final result` seems better.